src/pageNavigation.js

```diff
@@ -147,14 +147,33 @@ async function addListener(listener, {
 
     // Compile the match patterns for the listener
     const matchPatternSet = matching.createMatchPatternSet(matchPatterns);
-    // Register a content script for the listener
-    const contentScript = await browser.contentScripts.register({
-        matches: matchPatterns,
-        js: [{
-            file: pageNavigationContentScript
-        }],
-        runAt: "document_start"
-    });
+    // Only Firefox supports dynamic content script loading at this time.
+    let contentScript;
+    try {
+        // getBrowserInfo is Firefox-only as well.
+        const browserInfo = await browser.runtime.getBrowserInfo();
+        // Register a content script for the listener
+
+        if (browserInfo.name === "Firefox") {
```

Comment:
Maybe later we can have a util for this:

forBrowser(Browser.Firefox, () => /... browser specific code .../, () => /... optional else branch .../);

Reply:
Yes I think that would be easy to do, we had discussed doing a polyfill but it sounds like the current API isn't going to be polyfilled (maybe the new Chrome one will be implemented in Firefox instead), so I think this would be good for now.

This is going to be needed in other WS modules.

Reply:
Hm I don't mind doing this now, but I don't see a great place right now in this code for shared utilities :)

Reply:
OK I've created a `utils.js` and made a `registerContentScripts()` function, which should be easier to reuse.